parser/src/main/java/converter/BookDetailsToBookAssembler.java

```diff
@@ -116,15 +117,14 @@ private Book convert(BookDetails bookDetails) {
         return genres;
     }
 
-    // todo improve this method - iterate over getTags method
+
     private Set<Tag> retrieveTags(BookDetails bookDetails) {
-        Set<Tag> tags = new HashSet<>();
-        String tagString = tagMapper.getMap().get(bookDetails.getTags());
-        Tag tag = tagRepo.findByName(tagString);
-        if(tag != null) {
-            tags.add(tag);
+        if (bookDetails.getTags() == null) {

```

Comment:
Optional?
Or better add Optional to your last 'return' and there filter books without tags